<launch>

  <include file="$(find pylon_camera)/launch/pylon_camera_node.launch">
    <arg name="respawn" value="false"/>
    <arg name="config_file" value="$(find industrial_extrinsic_cal)/config/ACA1920_25gc.yaml"/>
  </include>

  <node pkg="cognex_demo" type="feature_detector_node" name="feature_detector" respawn="true">
    <param name="white_blobs" value="false"/>
    <param name="target_tracker/target_circle_dia" value="0.025"/>
    <param name="target_tracker/target_cols" value="4"/>
    <param name="target_tracker/target_rows" value="5"/>
    <param name="target_tracker/target_spacing" value="0.035"/>
    <param name="pylon_camera_node/max_area" value="80000"/>
  </node>

  <node pkg="cognex_demo" type="location_solver_node" name="location_solver" respawn="true"/>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find cognex_demo)/config/demo.rviz" required="true"/>

  <!-- <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="dynamic_reconfigure"/> -->

  <param name="pointcloud_file" value="$(find ros_cognex_demo)/support/plane_window.ply"/>
  <param name="pointcloud_frame" value="plane"/>
  <node pkg="cognex_demo" type="pointcloud_publisher" name="pointcloud_publisher"/>

  <node pkg="tf2_ros" type="static_transform_publisher" name="map_to_plane" args="0 0 1.0 0.707 0 0 0.707 map plane" respawn="true"/>

</launch>
